spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -101,6 +101,10 @@ public FirestoreTemplate(FirestoreStub firestore, String parent, FirestoreClassM
 		this.mappingContext = mappingContext;
 	}
 
+	public <T> FirestoreReactiveOperations operationsWithParent(T parent) {
```

Comment:
I think you should add to docs how you intend users to use this method. I am imagining something like this:

```
@Autowired
FirestoreTemplate template;

template.withRoot(user1).delete(...)
template.withRoot(user2).find(...)
```

Also I think we should consider renaming it from `operationsWithParent` to something like: `withCollectionRoot(...)`, `withRoot(...)`, or `withParent(..)`, etc. The method name should be short and also be clear that we are doing our operations from a different parent other than `/`.

Reply:
the docs are there in the docs submodule